moveit_core/planning_scene/test/test_multi_threaded.cpp

```diff
@@ -0,0 +1,139 @@
+#include <moveit/utils/robot_model_test_utils.h>
+#include <moveit/planning_scene/planning_scene.h>
+#include <urdf_parser/urdf_parser.h>
+#include <gtest/gtest.h>
+#include <thread>
+
+#include <moveit/collision_detection/collision_common.h>
+#include <moveit/collision_detection/collision_env.h>
+#include <moveit/collision_detection/collision_detector_allocator.h>
+
+#include <urdf_parser/urdf_parser.h>
+
+const int TRIALS = 1000;
+const int THREADS = 2;
+
+bool runCollisionDetection(unsigned int id, unsigned int trials, const planning_scene::PlanningScene* scene,
+                           const robot_state::RobotState* state)
+{
+  collision_detection::CollisionRequest req;
+  collision_detection::CollisionResult res;
+  for (unsigned int i = 0; i < trials; ++i)
+  {
+    res.clear();
+    scene->checkCollision(req, res, *state);
+  }
+  return res.collision;
+}
+
+void runCollisionDetectionAssert(unsigned int id, unsigned int trials, const planning_scene::PlanningScene* scene,
+                                 const robot_state::RobotState* state, bool expected_result)
+{
+  ASSERT_EQ(expected_result, runCollisionDetection(id, trials, scene, state));
+}
+
+class CollisionDetectorThreadedTest : public testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    robot_model_ = moveit::core::loadTestingRobotModel("panda");
+    robot_model_ok_ = static_cast<bool>(robot_model_);
+    robot_state_.reset(new robot_state::RobotState(robot_model_));
+    planning_scene_.reset(new planning_scene::PlanningScene(robot_model_));
+  }
+
+  void TearDown() override
+  {
+  }
+
+  bool robot_model_ok_;
+
+  robot_model::RobotModelPtr robot_model_;
+
+  collision_detection::CollisionEnvPtr cenv_;
+
+  collision_detection::AllowedCollisionMatrixPtr acm_;
+
+  robot_state::RobotStatePtr robot_state_;
+
+  planning_scene::PlanningScenePtr planning_scene_;
+};
+
+TEST_F(CollisionDetectorThreadedTest, InitOk)
+{
+  ASSERT_TRUE(robot_model_ok_);
+}
+
+/** \brief Tests the FCL collision detector in multiple threads. */
+TEST_F(CollisionDetectorThreadedTest, FCLThreaded)
+{
+  std::vector<robot_state::RobotStatePtr> states;
+  std::vector<std::thread*> threads;
+  std::vector<bool> collisions;
+
+  for (unsigned int i = 0; i < THREADS; ++i)
+  {
+    robot_state::RobotState* state = new robot_state::RobotState(planning_scene_->getRobotModel());
+    collision_detection::CollisionRequest req;
+    state->setToRandomPositions();
+    state->update();
+    states.push_back(robot_state::RobotStatePtr(state));
+    collisions.push_back(runCollisionDetection(0, 1, planning_scene_.get(), state));
+  }
+
+  for (unsigned int i = 0; i < THREADS; ++i)
+    threads.push_back(new std::thread(
+        std::bind(&runCollisionDetectionAssert, i, TRIALS, planning_scene_.get(), states[i].get(), collisions[i])));
+
+  for (unsigned int i = 0; i < states.size(); ++i)
+  {
+    threads[i]->join();
+    delete threads[i];
+  }
+  ASSERT_TRUE(true);
```

Comment:
Is this a critical test case? ;)

Reply:
wups, leftover from the commit before. Removed now.

Reply:
@j-petit it's still there?

Reply:
Sorry, I forgot to push I think.. Now it's definitely fixed, I double checked :)